configure.ac

```diff
@@ -379,6 +379,27 @@ then
 fi
 AC_MSG_RESULT(["$MACHDEP"])
 
+dnl For cross compilation, we distinguish between "prefix" (where we install the
+dnl files) and "host_prefix" (where we expect to find the files at runtime)
+
+if test -z "$host_prefix"; then
+  AS_CASE([$ac_sys_system],
+    [Emscripten], [host_prefix=/],
+    [WASI], [host_prefix=/],
```

Comment:
I haven't tried the PR out, but won't this forcibly anchor a WASI build to mount `lib` at `/`?

Reply:
I presume the WASI situation is more akin to the iOS/Android case, where the runtime mount point depends on how/where the code is running, rather than knowing ahead of time that the code will be installed at a known fixed mount point?

Reply:
Well it seems to me it would be better to have it default to `/` or `/usr` than to `/buildmachine/path/to/checkout/of/cpython/cross-build/wasm32-wasi`. If it's only known at runtime, we can control that by setting the `PYTHONHOME` environment variable or from the C api by setting the path directly. This only controls the final fallback case for determining the `prefix`, so it would also be possible to fix it by giving a value for `program_name` that makes sense in the file system visible to Python. Leaving this set based on the install target on the build machine doesn't seem that likely to be helpful in my opinion.

But I would also be perfectly happy to leave it with the default value in all cases, it is just nice to have the option of overriding it with a `configure` argument.

https://github.com/python/cpython/blob/main/Modules/getpath.py#L149-L151

Reply:
> Well it seems to me it would be better to have it default to `/` or `/usr` than to `/buildmachine/path/to/checkout/of/cpython/cross-build/wasm32-wasi`.

That's entirely a checkout thing and not a release thing. For instance, to package up the code, I [copy that file to the right place](https://github.com/brettcannon/cpython-wasi-build/blob/2cc59493dca88fd1a78921c6632277792afb7cb3/.github/workflows/release.yml#L93-L94). So that's why I'm asking if it only impacts a checkout or not as you can currently mount the files anywhere inside the WASI host and everything works since Python looks from the file location up for `lib/python3.NN`. As long as that keeps working then I'm fine with the change.


> This only controls [the final fallback case for determining the `prefix`](https://github.com/python/cpython/blob/main/Modules/getpath.py#L149-L151), so it would also be possible to fix it by giving a value for `program_name` that makes sense in the file system visible to Python.

This makes it sound like it won't break the case I want to preserve since step 5 in `getpath.py` will override:

https://github.com/python/cpython/blob/f474391b26aa9208b44ca879f8635409d322f738/Modules/getpath.py#L143-L147

As such, I'm good with the change, although if this is to help with `wasi.py` can that also be updated as a test case in this PR?

Reply:
> As such, I'm good with the change, although if this is to help with wasi.py can that also be updated as a test case in this PR?

Yeah I'll try it out and see if it works the way I think.

Reply:
It does not work the way I thought: `wasi.py` is mounting the checkout root into the wasi file system and using step 3 from `getpath.py` to run Python from the source checkout directory. The source checkout is clean so it's necessary to adjust the `PYTHONPATH` so that we pick up the `sysconfigdata`.
https://github.com/python/cpython/blob/main/Modules/getpath.py#L128-L135

But if I set `--prefix={wasi_build_dir}`, run `make commoninstall` instead of `make all`, and mount the `wasi_build_dir/lib` to `/lib` then this change helps. I'll push these changes to this branch to `wasi.py` to demonstrate and we can either keep them or revert them as you prefer.